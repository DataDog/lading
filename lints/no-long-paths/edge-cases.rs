// Edge cases and boundary conditions

// Use statements with various formats (should all be excluded)
use {
    std::collections::{HashMap, BTreeMap},
    tokio::sync::broadcast::Sender,
    crate::generator::http::Server
};

use self::module::submodule::function;

// Complex expressions
fn complex_cases() {
    // Method chaining - only the path segments should trigger
    let result = some::long::path::to::function()
        .map(|x| x.process())
        .unwrap_or_default();

    // Nested calls
    let value = outer::function(inner::long::path::call());

    // Array/struct initialization
    let config = Config {
        field: some::module::Type::default(),
        other: valid_two::segments(),
    };

    // Turbofish syntax
    let parsed = some::parser::module::<String>::parse();
}

// Comments and strings should not trigger
fn strings_and_comments() {
    // This comment mentions some::long::path::here but shouldn't trigger
    let string = "some::long::path::in::string";
    let raw_string = r#"another::long::path::here"#;
}

// Macros
fn macro_cases() {
    println!("valid");
    some_macro!(with::long::path::args);
    format!("string with {}", some::path::variable);
}