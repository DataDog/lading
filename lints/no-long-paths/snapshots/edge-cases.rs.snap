error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
  ┌─ lints/no-long-paths/edge-cases.rs:7:5
  │
7 │     crate::generator::http::Server
  │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/edge-cases.rs:15:18
   │
15 │     let result = some::long::path::to::function()
   │                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/edge-cases.rs:15:18
   │
15 │     let result = some::long::path::to::function()
   │                  ^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/edge-cases.rs:20:33
   │
20 │     let value = outer::function(inner::long::path::call());
   │                                 ^^^^^^^^^^^^^^^^^^^^^^^

error[no-long-paths]: Found long path (3+ segments) - import a subset of this path
   ┌─ lints/no-long-paths/edge-cases.rs:24:16
   │
24 │         field: some::module::Type::default(),
   │                ^^^^^^^^^^^^^^^^^^^^^^^^^^^

Error: 5 error(s) found in code.
Help: Scan succeeded and found error level diagnostics in the codebase.
